### DISCUSSION TITLE
Problem with amd64 image

### DISCUSSION THREAD

**Discussion by busubus85** [ltsp/ltsp#285] [Q&A]
Hello,
Thank you for your great LTSP.
I am installing a structure that includes 32 clients and 1 workstation

Workstation: i7 8700, 16gb RAM, n.3 HD 1 Tb SSD
Client i5 9500T, ram 8gb, 500Gb HD

I've configured everything and I'm using the "vm images" mode. 
So I created the virtual machine with virtualbox, with fixed disk size, and using the .VMDK format.
With 32Bit images everything works correctly, but when I create the virtual machine with a 64 bit image the following error occurs:

root@ltsp:/home/ltsp/Scrivania# ltsp image linux664
Running: losetup -rP /dev/loop15 /srv/ltsp/images/linux664.img
Running: mount -t tmpfs -o mode=0755 tmpfs /tmp/tmp.MmGXn7JAhF/tmpfs
Running: mount -t vfat -o ro /dev/loop15p1 /tmp/tmp.MmGXn7JAhF/tmpfs/0/looproot
Running: mount -t overlay -o upperdir=/tmp/tmp.MmGXn7JAhF/tmpfs/0/up,lowerdir=/tmp/tmp.MmGXn7JAhF/tmpfs/0/looproot,workdir=/tmp/tmp.MmGXn7JAhF/tmpfs/0/work /tmp/tmp.MmGXn7JAhF/tmpfs /tmp/tmp.MmGXn7JAhF/root/
mount: /tmp/tmp.MmGXn7JAhF/root: wrong fs type, bad option, bad superblock on /tmp/tmp.MmGXn7JAhF/tmpfs, missing codepage or helper program, or other error.
LTSP command failed: mount -t overlay -o upperdir=/tmp/tmp.MmGXn7JAhF/tmpfs/0/up,lowerdir=/tmp/tmp.MmGXn7JAhF/tmpfs/0/looproot,workdir=/tmp/tmp.MmGXn7JAhF/tmpfs/0/work /tmp/tmp.MmGXn7JAhF/tmpfs /tmp/tmp.MmGXn7JAhF/root/
Aborting ltsp

has anyone had my same problem?
Are virtualbox vmdk amd64 images supported on ltsp clients?

**Comment by alkisg** [comment]
I think you just didn't symlink the vmdk image correctly and now it's trying to open a bad /srv/ltsp/images/linux664.img image.
What's the output of this command?

```shell
ls -lhaR /srv/ltsp
```

You're supposed to do for example this:

```shell
sudo ln -s "/home/username/VirtualBox VMs/x86_64/x86_64-flat.vmdk" /srv/ltsp/x86_64.img
```

I.e. /srv/ltsp/x86_64.img should be a symlink to the target vmdk, so that then `ltsp image x86_64` will properly find it.

**Comment by busubus85** [comment]
Dear alkisg,
first of all thank you for your support.

The procedure you have installed, I have seen it several times.

In my example, to make the procedure faster I simply changed the name of the linux664-flat.vmdk - >> linux664.img file. 
Then I copied it to /srv /ltsp/image/

The strange thing is that with the images I create on 32-bit Virtualbox, everything works! Instead with 64-bit images from this error.

I have tried different versions of Ubuntu ....


ltsp@ltsp:~$ ls -lhaR /srv/ltsp/
/srv/ltsp/:
totale 12K
drwxrwxrwx 3 root root 4,0K mag 29 13:29 .
drwxr-xr-x 5 root root 4,0K mag 22 22:39 ..
drwxrwxrwx 2 root root 4,0K mag 28 20:56 images

/srv/ltsp/images:
totale 45G
drwxrwxrwx 2 root root 4,0K mag 28 20:56 .
drwxrwxrwx 3 root root 4,0K mag 29 13:29 ..
-rw------- 1 ltsp ltsp  16G mag 22 23:05 linux64.img
-rwxrwxrwx 1 ltsp ltsp  15G mag 23 01:17 linux664.img
-rw-r--r-- 1 ltsp ltsp  15G mar  6 18:22 Ubuntu.img

**Comment by alkisg** [comment]
What is the output of this command?

```shell
sudo fdisk -l linux664.img
```

There's no problem with 64bit vmdk images, I have 10+ of them.

**Comment by busubus85** [comment]
ltsp@ltsp:/srv/ltsp/images$ fdisk -l linux664.img
Disk linux664.img: 14,12 GiB, 15157231616 bytes, 29603968 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: dos
Disk identifier: 0x8a8b7f6a

Dispositivo   Avvio   Start     Fine  Settori  Size Id Tipo
linux664.img1 *        2048  1050623  1048576  512M  b W95 FAT32
linux664.img2       1052670 29603839 28551170 13,6G  5 Esteso
linux664.img5       1052672 29603839 28551168 13,6G 83 Linux
ltsp@ltsp:/srv/ltsp/images$

**Comment by alkisg** [comment]
So you have MBR, yet FAT32 as the first partition? What kind of installer did that?! :)
LTSP automatically uses the first mountable partition. If your Linux is not there, you need to read the `ltsp ipxe` man page, section "advanced image sources", to point it to the second partition:

https://ltsp.org/man/ltsp-ipxe/#advanced-image-sources

It's simpler if you just move your Linux to the first partition inside the .vmdk.

Btw, if images are named specifically "x86_32.img" and "x86_64.img", then LTSP automatically selects the correct one for 32bit or 64bit clients.

Finally, you can `mount --bind` the .vmdk over the .img file, you don't need to copy it.

**Comment by busubus85** [comment]
Alkisg,
 maybe you made my day better by solving the problem.
I do all the tests you have recommended and tomorrow I will give you the result.
Thank you so much!

**Comment by busubus85** [comment]
Thanks alkisg, I want to thank you because you solved the problem. 
I did some tests, and I found that by creating images with a single partition everything performed correctly.
Thanks so much. I hope this post helps others.
